### PR TITLE
[FIX] account_edi: avoid data from other companies in the invoice

### PR DESCRIPTION
**Pull request by ThomasBeckers**
When creating a bill from a bill mail alias, we can end up with a tax or a product from another company
inside the bill. This can cause access issues afterward.

Here is a use case:
- Setup with 2 companies -> A and B
- Both companies have a 21% tax for purchase but the tax from company A has a lower sequence number
- Create an invoice in company A with a 21% tax and print the PDF
- Send the PDF by mail to company B
- The bill generated from facturx will have the purchase tax from company A
=> access error when it's opened

We use 'with_company' function with 'self.env.company' to imply a company to search into but this lead to
two issues when the generation comes from a mail alias.

1. 'self.env.company' may not be the company of the invoice
2. 'self.env.su' is set to True -> company constraints from the environment are not applied

This commit aims to change those two problems.

opw-2752673

### CODE REVIEW

**Comment by ThomasBeckers** [comment]
@vin-odoo This is the PR we talked about yesterday

File: addons/account_edi/models/account_edi_format.py [hunk 1]
```diff
@@ -486,9 +486,9 @@ def _update_invoice_from_attachment(self, attachment, invoice):
                 res = False
                 try:
                     if file_data['type'] == 'xml':
-                        res = edi_format.with_company(self.env.company)._update_invoice_from_xml_tree(file_data['filename'], file_data['xml_tree'], invoice)
+                        res = edi_format.sudo(False).with_company(invoice.company_id)._update_invoice_from_xml_tree(file_data['filename'], file_data['xml_tree'], invoice)
```
[the remark this answers is not in the document]
Comment (ThomasBeckers) [reply addons/account_edi/models/account_edi_format.py:489]: Thanks for your response :+1: 
Ok so we are oblige to add the company in the domain when we search those fields (tax, product, etc) ? What kind of things we need to access with sudo rights ?
In my case (facturx received from a mail), the journal is set on the invoice.

**Review by FlorianGilbert — APPROVED** [review]
@robodoo r+

**Comment by FlorianGilbert** [comment]
@robodoo r-
Asked by tbs, waiting for jco

File: addons/account_edi/models/account_edi_format.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jco-odoo) [reply addons/account_edi/models/account_edi_format.py:489]: Normally you don't need to access things with sudo rights as normally, the user who does this, can create invoices.  @ThomasBeckers

[the remark this answers is not in the document]
Comment (ThomasBeckers) [reply addons/account_edi/models/account_edi_format.py:489]: So doing `sudo(False)` is ok no ?

[the remark this answers is not in the document]
Comment (jco-odoo) [reply addons/account_edi/models/account_edi_format.py:489]: I don't think so.  What is sudo(False) supposed to do?  (except undo all the accesses to companies)

**Comment by jco-odoo** [comment]
@robodoo r+

**Pull request closed without merging**